firmware/shared/src/io/io_can.c

```diff
@@ -1,111 +1,113 @@
 #include "io_can.h"
-#include <string.h>
+#undef NDEBUG
 #include <assert.h>
```

Comment:
Why do you want non-debug asserts here? Should probably make a dedicated `always_assert`/`debug_assert` or something like that to make this more clear. Otherwise I feel like this'll get confused a lot.

Reply:
yes I agree. This is something I carried over from the old code.

Reply:
ok maybe leave a TODO that this is something we should do?

Reply:
I guess we could use `hw_error` for checks that we want to be run in non-debug builds too

Reply:
yeah: also i don't think we got always_assert in C so we would have to make that macro ourselves, as in copy pasting the assert macro except not checking NDEBUG.